src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/DotNetInvoker.ts

```diff
@@ -0,0 +1,103 @@
+﻿import { MethodOptions, DotnetMethodArgumentsList } from './MethodOptions';
+import { platform } from '../Environment';
+import { System_String } from '../Platform/Platform';
+import { getRegisteredFunction } from './RegisteredFunction';
+
+export function invokeDotNetMethod<T>(methodOptions: MethodOptions, ...args: any[]): (T | null) {
+    const method = platform.findMethod(
+        "Microsoft.AspNetCore.Blazor.Browser",
+        "Microsoft.AspNetCore.Blazor.Browser.Interop",
+        "JavaScriptInvoke",
+        "InvokeDotnetMethod");
+
+    const packedArgs = packArguments(args);
+
+    const serializedOptions = platform.toDotNetString(JSON.stringify(methodOptions));
+    const serializedArgs = platform.toDotNetString(JSON.stringify(packedArgs));
+    const serializedResult = platform.callMethod(method, null, [serializedOptions, serializedArgs]);
+
+    if (serializedResult !== null && serializedResult !== undefined && (serializedResult as any) !== 0) {
+        const result = JSON.parse(platform.toJavaScriptString(serializedResult as System_String));
+        return result;
+    }
+
+    return null;
+}
+
+function packArguments(args: any[]): DotnetMethodArgumentsList {
+    var result = {};
+    if (args.length == 0) {
+        return result;
+    }
+
+    if (args.length > 7) {
+        for (let i = 0; i < 7; i++) {
+            result[`argument${[i + 1]}`] = args[i];
+        }
+        result['argument8'] = packArguments(args.slice(7));
+    } else {
+        for (let i = 0; i < args.length; i++) {
+            result[`argument${[i + 1]}`] = args[i];
+        }
+    }
+
+    return result;
+}
+
+let globalId = 0;
+
+export function invokeDotNetMethodAsync<T>(methodOptions: MethodOptions, ...args: any[]): Promise<T | null> {
```

Comment:
This looks great!

Reply:
Yeah I agree, this looks pretty straightforward and polished so far.